File: backend/logger/logger.go
package logger

import (
	"encoding/json"
	"fmt"
	"log"
	"os"
)

type LogLevel string

const (
	DEBUG LogLevel = "DEBUG"
	INFO  LogLevel = "INFO"
	WARN  LogLevel = "WARN"
	ERROR LogLevel = "ERROR"
)

type Logger struct {
	level  LogLevel
	logger *log.Logger
}

func New() *Logger {
	levelStr := os.Getenv("LOG_LEVEL")
	if levelStr == "" {
		levelStr = "INFO"
	}

	level := LogLevel(levelStr)
	if os.Getenv("ENVIRONMENT") == "production" {
		level = INFO // 프로덕션에서는 DEBUG 로그 제거
	}

	return &Logger{
		level:  level,
		logger: log.New(os.Stdout, "", log.LstdFlags),
	}
}

func (l *Logger) shouldLog(level LogLevel) bool {
	levels := map[LogLevel]int{
		DEBUG: 0,
		INFO:  1,
		WARN:  2,
		ERROR: 3,
	}
	return levels[level] >= levels[l.level]
}

func (l *Logger) Debug(format string, v ...interface{}) {
	if l.shouldLog(DEBUG) {
		l.log("DEBUG", format, v...)
	}
}

func (l *Logger) Info(format string, v ...interface{}) {
	if l.shouldLog(INFO) {
		l.log("INFO", format, v...)
	}
}

func (l *Logger) Warn(format string, v ...interface{}) {
	if l.shouldLog(WARN) {
		l.log("WARN", format, v...)
	}
}

func (l *Logger) Error(format string, v ...interface{}) {
	if l.shouldLog(ERROR) {
		l.log("ERROR", format, v...)
	}
}

func (l *Logger) log(level, format string, v ...interface{}) {
	logEntry := map[string]interface{}{
		"level":   level,
		"message": fmt.Sprintf(format, v...),
	}

	if os.Getenv("ENVIRONMENT") == "production" {
		jsonData, _ := json.Marshal(logEntry)
		l.logger.Println(string(jsonData))
	} else {
		l.logger.Printf("[%s] %s", level, fmt.Sprintf(format, v...))
	}
}

File: js/utils/logger.js
/**
 * 로깅 유틸리티
 * 모든 로그 출력 비활성화
 */
class Logger {
  static isDev() {
    return false;
  }

  static log(...args) {
    // 로그 비활성화
  }

  static error(...args) {
    // 로그 비활성화
  }

  static warn(...args) {
    // 로그 비활성화
  }

  static info(...args) {
    // 로그 비활성화
  }

  static debug(...args) {
    // 로그 비활성화
  }

  /**
   * 그룹 로그
   */
  static group(...args) {
    // 로그 비활성화
  }

  static groupEnd() {
    // 로그 비활성화
  }

  /**
   * 조건부 로그
   */
  static logIf(condition, ...args) {
    // 로그 비활성화
  }
}

export default Logger;

